### ISSUE TITLE
[Bug]: py_test does not work with bazel coverage

### ISSUE THREAD

**Issue by jab** [aspect-build/rules_py#353]
### What happened?

Using `bazel coverage` with an aspect `py_test` target always results in no coverage data being collected.

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`: aspect 5.10.1

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file: Current latest releases. See repro linked below.

Language(s) and/or frameworks involved: Python

### How to reproduce

Please see https://github.com/jab/aspect-py-test-coverage-bug for a minimal reproduction.

**Comment by alexeagle** [comment]
https://github.com/aspect-build/rules_py/issues/303 seems related - I see you aren't using a `py_pytest_main` so I'm not sure if the workaround there is useful.

**Comment by jab** [comment]
I didn't use `py_pytest_main` so as to provide a more minimal reproduction. But I am interested in experimenting with `py_pytest_main` more at some point.

I did take a quick a look at #303, but since I prefer not to use `pytest-cov`, it didn't seem like it would apply to me. (I agree with the reasons in https://www.youtube.com/watch?v=sPgvHGkmd0U for preferring to use coverage.py to invoke my test runner directly, as `bazel coverage` does with py_test targets.)

**Comment by alexeagle** [comment]
I happened to reproduce this today while working on the Python course and debugging for a client.

**Comment by garymm** [comment]
I'm seeing no coverage data on 1.3.0, not using pytest. Using absltest. Do we have to use pytest to get coverage?
I prefer to not use pytest because it makes test startup noticably slower.

**Comment by alexeagle** [comment]
@garymm yes you're right, my change closing this issue is specific to pytest.

**Comment by ctcjab** [comment]
Heads up for anyone landing here: this was fixed by #492. The runner now imports `coverage` and writes lcov directly when `COVERAGE_MANIFEST` is set.

The trap that still bites people is that the runner falls back to a no-op when the `coverage` package isn't in the test target's `deps`. The only signal is a one-line warning printed to stdout from `pytest.py.tmpl`:

> `WARNING: python coverage setup failed. Do you need to include the 'coverage' package as a dependency of py_pytest_main? No module named 'coverage'`

Bazel still happily emits an empty `_coverage_report.dat` (just `SF:`/`LF:0`/`LH:0` stanzas), which looks identical to the original bug report and sends you back here.

Two small things that could help:

1. **Document the `coverage` requirement** in `docs/` (the only place it's mentioned today is the dep list in `examples/pytest/BUILD.bazel`).
2. **Make the warning louder** — print to stderr, or fail the coverage run outright when `COVERAGE_MANIFEST` is set but `import coverage` fails. Right now an empty report is indistinguishable from a real "no lines hit" result.

Either way, this issue itself can probably be closed as fixed by #492.
